### DISCUSSION TITLE
Teensy 4.1: Notes transform in CC under Mux

### DISCUSSION THREAD

**Discussion by antomexita** [tttapa/Control-Surface#972] [General]
Hi, 
i have a problem with this code:

```cpp
#include <Control_Surface.h>
USBDebugMIDI_Interface midi = 115200;
CD74HC4067 drumpad = {
  20,
  { 2, 3, 4, 5 },
};
CCButton DP[]{
  { drumpad.pin(8), { MIDI_Notes::C(4), Channel_1 } },
  { drumpad.pin(9), { MIDI_Notes::D(4), Channel_2 } },
  { drumpad.pin(10), { MIDI_Notes::E(4), Channel_3 } },
  { drumpad.pin(11), { MIDI_Notes::F_(4), Channel_4 } },
  { drumpad.pin(12), { MIDI_Notes::G(4), Channel_5 } },
  { drumpad.pin(13), { MIDI_Notes::A(4), Channel_6 } },
  { drumpad.pin(14), { MIDI_Notes::B(4), Channel_7 } },
  { drumpad.pin(15), { MIDI_Notes::C(5), Channel_8 } },
  { drumpad.pin(7), { MIDI_Notes::D(5), Channel_9 } },
}
void setup() {
  drumpad.pinMode(0, INPUT_PULLUP);
  Control_Surface.begin();
}
void loop() {
  Control_Surface.loop();
}
```

When upload on Teensy 4.1 the result is:

19:52:46.412 -> Control Change   Channel: 4	Data 1: 0x41	Data 2: 0x7f
19:52:46.585 -> Control Change   Channel: 4	Data 1: 0x41	Data 2: 0x00
19:52:47.559 -> Control Change   Channel: 2	Data 1: 0x3e	Data 2: 0x7f
19:52:47.762 -> Control Change   Channel: 2	Data 1: 0x3e	Data 2: 0x00
19:54:06.838 -> Control Change   Channel: 3	Data 1: 0x40	Data 2: 0x7f
19:54:07.043 -> Control Change   Channel: 3	Data 1: 0x40	Data 2: 0x00
19:54:07.982 -> Control Change   Channel: 5	Data 1: 0x43	Data 2: 0x7f
19:54:08.120 -> Control Change   Channel: 5	Data 1: 0x43	Data 2: 0x00
19:54:08.821 -> Control Change   Channel: 6	Data 1: 0x45	Data 2: 0x7f
19:54:08.954 -> Control Change   Channel: 6	Data 1: 0x45	Data 2: 0x00
19:54:09.638 -> Control Change   Channel: 9	Data 1: 0x4a	Data 2: 0x7f
19:54:09.818 -> Control Change   Channel: 9	Data 1: 0x4a	Data 2: 0x00
19:54:10.406 -> Control Change   Channel: 8	Data 1: 0x48	Data 2: 0x7f
19:54:10.609 -> Control Change   Channel: 8	Data 1: 0x48	Data 2: 0x00
19:54:11.056 -> Control Change   Channel: 7	Data 1: 0x47	Data 2: 0x7f
19:54:11.290 -> Control Change   Channel: 7	Data 1: 0x47	Data 2: 0x00

I dont understand where is the error.
Thanks in advice

**Comment by antomexita** [comment]
Another problem  (just discovered) is that when I upload the NoteLEDBar.ino example i need to send C3 note instead of C4 to turn on the leds.

**Comment by tttapa** [comment]
The `CCButton` class will always send Control Change messages. You should use a `NoteButton` if you want to send Note On/Off messages.

Control Surface uses the Scientific Pitch Notation system, some MIDI programs use a different convention, see https://en.wikipedia.org/wiki/Scientific_pitch_notation#Similar_systems. 
I've now clarified this in the documentation: [Control Surface: MIDI_Notes Namespace Reference](https://tttapa.github.io/Control-Surface-doc/Doxygen/d7/d78/namespaceMIDI__Notes.html)

**Reply by antomexita** [reply]
Hi, I've solved the mux problem but after days I still don't solve the problem of the wrong octave. I connect the teensy with Garage Band, Logic and PoketMidi (like MidiOX) and all softwares read notes 1 octave under (ej C4 on the Arduino sketch will be read C3). What test can I do to solve the problem? Anyway thanks for the good work

**Reply by tttapa** [reply]
There are simply different conventions to define the octave number. In some DAWs, middle C is C3, in others it's C5, in even others it's C4. You can find many discussions about this online: https://www.reddit.com/r/musictheory/comments/a17he3/do_yall_use_c3_or_c4_as_middle_c_and_why_isnt

Control Surface uses the convention that middle C is C4, which is the standard used in music theory. Your DAW seems to use C3 for middle C.  
The solution is simple though: add one to the octave number in your DAW, or subtract one from the octave number on the Arduino.
